pkg/kube/krt/collection.go

```diff
@@ -79,25 +109,31 @@ func (i dependencyState[I]) changedInputKeys(sourceCollection collectionUID, eve
 		// Naively, we can look through every item in this collection and check if it matches the filter. However, this is
 		// inefficient, especially when the dependency changes frequently and the collection is large.
 		// Where possible, we utilize the reverse-indexing to get the precise list of potentially changed objects.
-		if extractor, f := i.indexedDependenciesExtractor[sourceCollection]; f {
-			// We have a reverse index
-			for _, item := range ev.Items() {
-				// Find all the reverse index keys for this object. For each key we will find impacted input objects.
-				keys := extractor(item)
-				for _, key := range keys {
-					for iKey := range i.indexedDependencies[indexedDependency{id: sourceCollection, key: key}] {
-						if changedInputKeys.Contains(iKey) {
-							// We may have already found this item, skip it
-							continue
-						}
-						dependencies := i.objectDependencies[iKey]
-						if changed := objectChanged(dependencies, sourceCollection, ev, true); changed {
-							changedInputKeys.Insert(iKey)
+		foundAny := false
+		for _, idxTypes := range allIndexedDependencyTypes {
```

Comment:
nit I missed the fist time through:
`idxType` not Types would be preferred I think